### PR TITLE
[Cleanup][A-13] clean some VarType for test

### PR DESCRIPTION
**Pull request by co63oc**
### PR types

Others 
### PR changes

Others 
### Description

https://github.com/PaddlePaddle/Paddle/issues/61385

### CODE REVIEW

File: python/paddle/static/amp/fp16_lists.py [hunk 1]
```diff
@@ -72,9 +72,9 @@ def get_low_precision_dtypestr(dtype):
     if isinstance(dtype, str):
         return check_amp_dtype(dtype)
     elif isinstance(dtype, core.VarDesc.VarType):
-        if dtype == core.VarDesc.VarType.FP16:
+        if dtype == paddle.float16:
             return "float16"
-        elif dtype == core.VarDesc.VarType.BF16:
+        elif dtype == paddle.bfloat16:
             return "bfloat16"
```
Comment (SigureMo) [inline python/paddle/static/amp/fp16_lists.py]: 这里做一下兼容性处理吧，`isinstance(dtype, (core.VarDesc.VarType, core.DataType))`

File: python/paddle/static/nn/common.py [hunk 2]
```diff
@@ -343,7 +343,7 @@ def instance_norm(
     dtype = helper.input_dtype()
 
     # use fp32 for in parameter
-    if dtype == paddle.framework.core.VarDesc.VarType.FP16:
+    if dtype == paddle.float16:
         dtype = paddle.framework.core.VarDesc.VarType.FP32
```
Comment (SigureMo) [inline python/paddle/static/nn/common.py]: 这里相应地修改 dtype

File: python/paddle/static/nn/common.py [hunk 3]
```diff
@@ -2765,7 +2765,7 @@ def batch_norm(
     dtype = helper.input_dtype()
 
     # use fp32 for bn parameter
-    if dtype == core.VarDesc.VarType.FP16 or dtype == core.VarDesc.VarType.BF16:
+    if dtype == paddle.float16 or dtype == paddle.bfloat16:
         dtype = core.VarDesc.VarType.FP32
```
Comment (SigureMo) [inline python/paddle/static/nn/common.py]: 同上

**Comment by luotao1** [comment]
@SigureMo 可以再review一下

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**